src/ssl.c

```diff
@@ -44374,34 +44383,25 @@ int wolfSSL_i2d_X509_NAME(WOLFSSL_X509_NAME* name, unsigned char** out)
         return BUFFER_E;
     }
 
-    /* check if using buffer passed in */
-    if (*out == NULL) {
-        *out = local = (unsigned char*)XMALLOC(totalBytes + idx, NULL,
-                DYNAMIC_TYPE_OPENSSL);
-        if (*out == NULL) {
-            return MEMORY_E;
-        }
-    }
-
-    /* header */
-    idx = SetSequence(totalBytes, temp);
-    if (totalBytes + idx > ASN_NAME_MAX) {
+    if (out == NULL) {
+        /* If out is NULL, caller just wants length. */
+        totalBytes += idx;
 #ifdef WOLFSSL_SMALL_STACK
         XFREE(names, NULL, DYNAMIC_TYPE_TMP_BUFFER);
 #endif
-        WOLFSSL_MSG("Total Bytes is greater than ASN_NAME_MAX");
-        return BUFFER_E;
+        return totalBytes;
     }
-
-    /* check if using buffer passed in */
-    if (*out == NULL) {
-        *out = local = (unsigned char*)XMALLOC(totalBytes + idx, name->heap,
-                DYNAMIC_TYPE_OPENSSL);
+    else {
```

Comment:
No need for else.

Reply:
I meant no need to have 'else' when you return.
Still need the code in the else!

Reply:
I think the code is still there. Just GitHub being confusing with the diff.

Reply:
I checked out the code and saw that it is still there.